cmake/NanoBindMinimal.cmake

```diff
@@ -0,0 +1,16 @@
+# =============================================================================
+# minimal nanobind interface
+# =============================================================================
+
+set(NB_DIR ${PROJECT_SOURCE_DIR}/external/nanobind)
+
+function(make_nanobind_target TARGET_NAME PYINC)
+  add_library(${TARGET_NAME} STATIC ${NB_DIR}/src/nb_combined.cpp)
+  target_include_directories(${TARGET_NAME} SYSTEM PUBLIC ${NB_DIR}/include)
+  target_include_directories(${TARGET_NAME} SYSTEM PRIVATE ${NB_DIR}/ext/robin_map/include ${PYINC})
+  target_compile_options(${TARGET_NAME} PRIVATE -fno-strict-aliasing)
```

Comment:
This seems like an GCC/Clang flag, will it work for other compiers?

Reply:
@ferdonline I didn't touch this part, can you give us some comments why is it needed?

Reply:
It is taken from a comment in the original Nanobind cmake. But good point, we should have compiler guards, like the original:
```
if (MSVC)
    # Do not complain about vsnprintf
    target_compile_definitions(${TARGET_NAME} PRIVATE -D_CRT_SECURE_NO_WARNINGS)
  else()
    # Generally needed to handle type punning in Python code
    target_compile_options(${TARGET_NAME} PRIVATE -fno-strict-aliasing)
  endif()
```